mlflow/gateway/providers/base.py

```diff
@@ -16,26 +16,44 @@ class BaseProvider(ABC):
     SUPPORTED_ROUTE_TYPES: Tuple[str, ...]
 
     def __init__(self, config: RouteConfig):
+        if not hasattr(self, "NAME"):
+            raise TypeError(f"'NAME' is not defined for {self.__class__.__name__}")
```

Comment:
I'm not a big fan of this. Is there a way to enforce the `NAME` attribute?

Reply:
Can we use an abstract property instead?

Reply:
or just remove this.

Reply:
i think abstract property can do. If we don't do any validation, we'll need to rely solely on tests or get runtime errors

Reply:
Let's use abstract property :)

Reply:
sure, let's use an empty string :)